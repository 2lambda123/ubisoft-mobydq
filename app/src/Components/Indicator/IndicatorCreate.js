import React from 'react';
import { Query } from "react-apollo";
import gql from "graphql-tag";
import { styles } from './../../styles/baseStyles';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import DataTable from '../Dashboard/DataTable'
import SimpleForm from '../SimpleForm/SimpleForm'

class IndicatorCreate extends React.Component {
  render() {
    const { classes } = this.props;
    const IndicatorList = () => (
      <Query
        query={gql`
          {
            allIndicators(first:2, offset: 1) {
              nodes{
                id
                name
                description
                executionOrder
                flagActive
                createdDate
                updatedDate
                indicatorTypeId
              }
            }
          }
        `}
      >
        {({ loading, error, data }) => {
          if (loading) return <p>Loading...</p>;
          if (error) return <p>Error :(</p>;
          return(
            <SimpleForm/>
          );
        }}
      </Query>
    );
    return (
      <React.Fragment>
        <div className={classes.appBarSpacer} />
        <Typography variant="indicator-create-form" gutterBottom className={classes.formContainer}>
          <SimpleForm/>
        </Typography>
      </React.Fragment>
     )
  }
}
export default withStyles(styles)(IndicatorCreate);