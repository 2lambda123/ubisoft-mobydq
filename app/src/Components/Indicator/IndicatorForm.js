import React from 'react';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import TextInput from './../FormInput/TextInput';
import SelectInput from './../FormInput/SelectInput';
import SwitchInput from './../FormInput/SwitchInput';
import SaveButton from './../FormInput/SaveButton';
import ExecuteButton from './../FormInput/ExecuteButton';

const IndicatorFormFields = (props) => {
  const {
    data,
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting
  } = props;
  return <form onSubmit={handleSubmit} style={{ 'marginLeft': '50px' }}>
    <div style={{ 'marginTop': '10px', 'marginBottom': '30px' }}>
      <SaveButton disabled={isSubmitting} />
      <ExecuteButton disabled={isSubmitting} />
    </div>
    <div>
      <TextInput
        id="name"
        label="Name"
        helperText=""
        placeholder=""
        touched={touched.name}
        error={touched.name && errors.name}
        value={values.name}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="description"
        label="Description"
        helperText=""
        placeholder=""
        touched={touched.description}
        error={touched.description && errors.description}
        value={values.description}
        onChange={handleChange}
        onBlur={handleBlur}
        multiline={true}
        rows={5}
      />
    </div>
    <div>
      <SelectInput
        id="indicatorTypeId"
        label="Type"
        items={data.allIndicatorTypes.nodes}
        touched={touched.indicatorTypeId}
        error={touched.indicatorTypeId && errors.indicatorTypeId}
        value={values.indicatorTypeId}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <SelectInput
        id="indicatorGroupId"
        label="Group"
        items={data.allIndicatorGroups.nodes}
        touched={touched.indicatorGroupId}
        error={touched.indicatorGroupId && errors.indicatorGroupId}
        value={values.indicatorGroupId}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="executionOrder"
        label="Execution Order"
        helperText=""
        numeric={true}
        touched={touched.executionOrder}
        error={touched.executionOrder && errors.executionOrder}
        value={values.executionOrder}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <SwitchInput
        id="flagActive"
        label="Active"
        touched={touched.flagActive}
        error={touched.flagActive && errors.flagActive}
        checked={values.flagActive}
        value="flagActive"
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="createdDate"
        label="Created Date"
        helperText=""
        placeholder=""
        touched={touched.createdDate}
        error={touched.createdDate && errors.createdDate}
        value={values.createdDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="createdBy"
        label="Created By"
        helperText=""
        placeholder=""
        touched={touched.createdBy}
        error={touched.createdBy && errors.createdBy}
        value={values.createdBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="updatedDate"
        label="Updated Date"
        helperText=""
        placeholder=""
        touched={touched.updatedDate}
        error={touched.updatedDate && errors.updatedDate}
        value={values.updatedDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="updatedBy"
        label="Updated By"
        helperText=""
        placeholder=""
        touched={touched.updatedBy}
        error={touched.updatedBy && errors.updatedBy}
        value={values.updatedBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
  </form>;
};

const formikEnhancer = withFormik({
  'validationSchema': Yup.object().shape({
    'indicatorTypeId': Yup.number().integer()
      .min(1, 'You need to select an indicator type')
      .required('You need to select an indicator type'),
    'indicatorGroupId': Yup.number().integer()
      .min(1, 'You need to select an indicator group')
      .required('You need to select an indicator group'),
    'executionOrder': Yup.number().integer()
      .min(0, 'Execution order has to be a non-negative integer')
      .required('You need to input execution order.'),
    'name': Yup.string()
      .required('Name cannot be blank'),
    'description': Yup.string()
      .required('Description cannot be blank')
  }),

  'mapPropsToValues': (props) => props.initialFieldValues === null
    ? {
        'name': '',
        'description': '',
        'executionOrder': 0,
        'indicatorTypeId': 0,
        'indicatorGroupId': 0,
        'flagActive': false
      }
    : {
        'name': props.initialFieldValues.name,
        'description': props.initialFieldValues.description,
        'executionOrder': props.initialFieldValues.executionOrder,
        'indicatorTypeId': props.initialFieldValues.indicatorTypeId,
        'indicatorGroupId': props.initialFieldValues.indicatorGroupId,
        'flagActive': props.initialFieldValues.flagActive,
        'createdDate': props.initialFieldValues.createdDate,
        'createdBy': props.initialFieldValues.userByCreatedById.email,
        'updatedDate': props.initialFieldValues.updatedDate,
        'updatedBy': props.initialFieldValues.userByUpdatedById.email
      },
  'handleSubmit': (payload, { props, setSubmitting }) => {
    setSubmitting(false);
    let variables;
    if (props.initialFieldValues === null) {
      variables = { 'indicator': payload };
    } else {
      variables = { 'indicatorPatch': payload, 'id': props.initialFieldValues.id };
    }
    props.mutate({
      variables
    });
  },
  'displayName': 'IndicatorForm'
});


const EnhancedIndicatorForm = formikEnhancer(IndicatorFormFields);

export default EnhancedIndicatorForm;
