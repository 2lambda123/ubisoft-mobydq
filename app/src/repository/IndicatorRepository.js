import gql from 'graphql-tag';

class IndicatorRepository {
  static getListPage(pageNumber, pageSize) { // eslint-disable-line no-unused-vars
    return gql`
      {
        allIndicators {
          nodes {
            id
            name
            indicatorTypeId
            indicatorGroupId
            executionOrder
            flagActive
            updatedDate
          }
        }
      }
    `
  }

  static getFormDropdownData() {
    return gql`
      {
        allIndicatorTypes {
          nodes {
            id
            name
          }
        }
        allIndicatorGroups {
          nodes {
            id
            name
          }
        }
      }
    `
  }

  static display() {
    return gql`
        query getIndicator($id: Int!) {
          indicatorById(id: $id) {
            id
            name
            description
            indicatorTypeId
            indicatorGroupId
            executionOrder
            flagActive
            updatedDate
          }
        }
    `
  }

  static insert() {
    return gql`
      mutation addNewIndicator($indicator: IndicatorInput!) {
        createIndicator(input: {indicator: $indicator}) {
          indicator {
            id
            name
            description
          }
        }
      }
    `
  }

  static getIndicatorToUpdate(id) {
    return gql`
      {
        indicatorById(id:` + id + `) {
          id
          name
          description
          executionOrder
          flagActive
          createdDate
          updatedDate
          indicatorTypeId
        }
      }
    `
  }

  static update() {
    return gql`
      mutation updateIndicator($id: Int!, $indicatorPatch: IndicatorPatch!) {
        updateIndicatorById(input: {id: $id, indicatorPatch: $indicatorPatch}) {
          indicator {
            id
          }
        }
      }
    `
  }
}

export default IndicatorRepository;