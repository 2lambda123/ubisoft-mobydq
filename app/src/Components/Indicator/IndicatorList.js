import React from 'react';

import { connect } from 'react-redux';
import { setIndicatorPage, setIndicatorRowsPerPage, setIndicatorRowTotal } from './../../actions/indicatorList';

import { Query } from 'react-apollo';
import IndicatorRepository from './../../repository/IndicatorRepository';
import { GraphQLError } from './../Error/GraphQLError';

import ListTable from '../ListTable/ListTable';
import LinkButton from './../../Components/FormInput/LinkButton';

class IndicatorList extends React.Component {
  render() {
    return (
      <Query
        query={IndicatorRepository.getListPage()}
        variables={{ 'first': this.props.indicatorRowsPerPage, 'offset': this.props.indicatorPage * this.props.indicatorRowsPerPage }}
        fetchPolicy={this.props.refetch ? 'cache-and-network' : 'cache-first'}
      >
        {({ loading, error, data }) => {
          if (loading) {
            return <p>Loading...</p>;
          }
          if (error) {
            return <GraphQLError error={error}/>;
          }
          this.props.setIndicatorRowTotal(data.allIndicators.totalCount);
          return (
            <div>
              <div style={{ 'float': 'left', 'marginLeft': '60px' }}>
                Indicators
              </div>
              <div style={{ 'float': 'right' }}>
                <LinkButton label="Create" type="Create" color="primary"
                  variant="contained" to={'/indicator/new'}/>
              </div>
              <ListTable
                data={data.allIndicators.nodes}
                buttons={[
                  { 'function': 'edit', 'parameter': '/indicator-group' },
                  { 'function': 'delete', 'parameter': this._buildDeleteParam() }
                ]}
                footerParams={this._buildFooterParam()}
              />
            </div>
          );
        }}
      </Query>
    );
  }

  _buildDeleteParam() {
    return {
      'page': this.props.indicatorPage,
      'rowTotal': this.props.indicatorRowTotal,
      'rowsPerPage': this.props.indicatorRowsPerPage,
      'setPage': this.props.setIndicatorPage,
      'repository': IndicatorRepository
    };
  }

  _buildFooterParam() {
    return {
      'page': this.props.indicatorPage,
      'rowTotal': this.props.indicatorRowTotal,
      'rowsPerPage': this.props.indicatorRowsPerPage,
      'setPage': this.props.setIndicatorPage,
      'setRowsPerPage': this.props.setIndicatorRowsPerPage
    };
  }
}

const mapStateToProps = (state) => ({
  'indicatorPage': state.indicatorPage,
  'indicatorRowsPerPage': state.indicatorRowsPerPage,
  'indicatorRowTotal': state.indicatorRowTotal
});

const mapDispatchToProps = (dispatch) => ({
  'setIndicatorPage': (page) => dispatch(setIndicatorPage(page)),
  'setIndicatorRowsPerPage': (rowsPerPage) => dispatch(setIndicatorRowsPerPage(rowsPerPage)),
  'setIndicatorRowTotal': (rowTotal) => dispatch(setIndicatorRowTotal(rowTotal))
});

export default connect(mapStateToProps, mapDispatchToProps)(IndicatorList);

