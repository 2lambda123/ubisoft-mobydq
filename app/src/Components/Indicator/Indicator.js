import React from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import { styles } from './../../styles/baseStyles';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import RouterButton from './../../Components/FormInput/RouterButton'
import IndicatorList from './IndicatorList'
import IndicatorCreate from './IndicatorForm'

class Indicator extends React.Component {
  render() {
    const { classes } = this.props;
    const { match } = this.props;
    return (
      <React.Fragment>
        <div className={classes.appBarSpacer} />
        <Typography variant='display1' gutterBottom className={classes.chartContainer}>
           <Route path={`${match.url}/new`} component={IndicatorCreate} />
           <Route
             exact
             path={match.url}
             render={() => <IndicatorList refetch />}
            />
        </Typography>
      </React.Fragment>
    )
  }
}
export default withStyles(styles)(Indicator);