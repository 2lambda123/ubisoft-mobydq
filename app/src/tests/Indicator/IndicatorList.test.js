import React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';

import IndicatorRepository from './../../repository/IndicatorRepository';
import IndicatorList from '../../Components/Indicator/IndicatorList';
import Indicator from "../../Components/Indicator/Indicator";

describe('IndicatorList unit test', () => {
  let wrapper;
  let query = IndicatorRepository.getListPage(1, 10);

  let mocks = [
  {
    request: {
      query: query
    },
    result: {
       data: {
        allIndicators: {
          nodes: [
            {
              id: 1,
              name: "dq_example_completeness_indicator",
              indicatorTypeId: 1,
              indicatorGroupId: 1,
              executionOrder: 1,
              flagActive: true,
              updatedDate: "2018-10-06T05:44:17.45341"
            },
            {
              id: 2,
              name: "dq_example_freshness_indicator",
              indicatorTypeId: 2,
              indicatorGroupId: 1,
              executionOrder: 1,
              flagActive: true,
              updatedDate: "2018-10-06T05:44:17.45341"
            },
            {
              id: 3,
              name: "dq_example_latency_indicator",
              indicatorTypeId: 3,
              indicatorGroupId: 1,
              executionOrder: 1,
              flagActive: true,
              updatedDate: "2018-10-06T05:44:17.45341"
            },
            {
              id: 4,
              name: "dq_example_validity_indicator",
              indicatorTypeId: 4,
              indicatorGroupId: 1,
              executionOrder: 1,
              flagActive: true,
              updatedDate: "2018-10-06T05:44:17.45341"
            }
          ]
        }
      }
    },
  },
];

  beforeEach(() => {
    wrapper = shallow(
      <MockedProvider mocks={mocks} addTypename={true}>
        <IndicatorList/>
      </MockedProvider>
    )
  });

  it('renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('loads the table properly', () => {
     expect(wrapper).toEqual(4);
     expect(wrapper.find('th').length).toEqual(7);
     expect(wrapper.find('td').length).toEqual(28);
     expect(wrapper.find('th').at(0)).toEqual('Id');
     expect(wrapper.find('td').at(0)).toEqual('1');
     expect(wrapper.find('td').at(11)).toEqual('1');
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('IndicatorList unit test', () => {
  let wrapper;
  let query = IndicatorRepository.getListPage(1, 10);

  let mocks = [
  {
    request: {
      query: query
    },
    result: {
       data: {
        allIndicators: {
          nodes: [
            {
              id: 1,
              name: "dq_example_completeness_indicator",
              indicatorTypeId: 1,
              indicatorGroupId: 1,
              executionOrder: 1,
              flagActive: true,
              updatedDate: "2018-10-06T05:44:17.45341"
            },
            {
              id: 2,
              name: "dq_example_freshness_indicator",
              indicatorTypeId: 2,
              indicatorGroupId: 1,
              executionOrder: 1,
              flagActive: true,
              updatedDate: "2018-10-06T05:44:17.45341"
            },
            {
              id: 3,
              name: "dq_example_latency_indicator",
              indicatorTypeId: 3,
              indicatorGroupId: 1,
              executionOrder: 1,
              flagActive: true,
              updatedDate: "2018-10-06T05:44:17.45341"
            },
            {
              id: 4,
              name: "dq_example_validity_indicator",
              indicatorTypeId: 4,
              indicatorGroupId: 1,
              executionOrder: 1,
              flagActive: true,
              updatedDate: "2018-10-06T05:44:17.45341"
            }
          ]
        }
      }
    },
  },
];

  //todo figure out why enzyme mount hangs the tests
  beforeEach(() => {
    wrapper = shallow(
      <MockedProvider mocks={mocks} addTypename={true}>
        <IndicatorList/>
      </MockedProvider>
    )
  });

  it('renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('loads the table properly', () => {
     expect(wrapper).toEqual(4);
     expect(wrapper.find('th').length).toEqual(7);
     expect(wrapper.find('td').length).toEqual(28);
     expect(wrapper.find('th').at(0)).toEqual('Id');
     expect(wrapper.find('td').at(0)).toEqual('1');
     expect(wrapper.find('td').at(11)).toEqual('1');
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

