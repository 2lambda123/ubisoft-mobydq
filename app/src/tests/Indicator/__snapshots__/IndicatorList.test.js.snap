// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorList unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MockedProvider
    addTypename={true}
    mocks={
      Array [
        Object {
          "request": Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "indicatorRange",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "first",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "first",
                              },
                            },
                          },
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "offset",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "offset",
                              },
                            },
                          },
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "orderBy",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "orderBy",
                              },
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "allIndicators",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "totalCount",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "nodes",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "name",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "indicatorTypeId",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "indicatorGroupId",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "executionOrder",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "flagActive",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "updatedDate",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [
                    Object {
                      "defaultValue": undefined,
                      "directives": Array [],
                      "kind": "VariableDefinition",
                      "type": Object {
                        "kind": "NonNullType",
                        "type": Object {
                          "kind": "NamedType",
                          "name": Object {
                            "kind": "Name",
                            "value": "Int",
                          },
                        },
                      },
                      "variable": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "first",
                        },
                      },
                    },
                    Object {
                      "defaultValue": undefined,
                      "directives": Array [],
                      "kind": "VariableDefinition",
                      "type": Object {
                        "kind": "NonNullType",
                        "type": Object {
                          "kind": "NamedType",
                          "name": Object {
                            "kind": "Name",
                            "value": "Int",
                          },
                        },
                      },
                      "variable": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "offset",
                        },
                      },
                    },
                    Object {
                      "defaultValue": undefined,
                      "directives": Array [],
                      "kind": "VariableDefinition",
                      "type": Object {
                        "kind": "ListType",
                        "type": Object {
                          "kind": "NonNullType",
                          "type": Object {
                            "kind": "NamedType",
                            "name": Object {
                              "kind": "Name",
                              "value": "IndicatorsOrderBy",
                            },
                          },
                        },
                      },
                      "variable": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "orderBy",
                        },
                      },
                    },
                  ],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 402,
                "start": 0,
              },
            },
          },
          "result": Object {
            "data": Object {
              "allIndicators": Object {
                "nodes": Array [
                  Object {
                    "executionOrder": 1,
                    "flagActive": true,
                    "id": 1,
                    "indicatorGroupId": 1,
                    "indicatorTypeId": 1,
                    "name": "dq_example_completeness_indicator",
                    "updatedDate": "2018-10-06T05:44:17.45341",
                  },
                  Object {
                    "executionOrder": 1,
                    "flagActive": true,
                    "id": 2,
                    "indicatorGroupId": 1,
                    "indicatorTypeId": 2,
                    "name": "dq_example_freshness_indicator",
                    "updatedDate": "2018-10-06T05:44:17.45341",
                  },
                  Object {
                    "executionOrder": 1,
                    "flagActive": true,
                    "id": 3,
                    "indicatorGroupId": 1,
                    "indicatorTypeId": 3,
                    "name": "dq_example_latency_indicator",
                    "updatedDate": "2018-10-06T05:44:17.45341",
                  },
                  Object {
                    "executionOrder": 1,
                    "flagActive": true,
                    "id": 4,
                    "indicatorGroupId": 1,
                    "indicatorTypeId": 4,
                    "name": "dq_example_validity_indicator",
                    "updatedDate": "2018-10-06T05:44:17.45341",
                  },
                ],
              },
            },
          },
        },
      ]
    }
  >
    <Connect(IndicatorList) />
  </MockedProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(IndicatorList) />,
      "client": ApolloClient {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "clientAwareness": Object {},
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "typeDefs": undefined,
        "version": "2.5.0",
        "watchQuery": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(IndicatorList) />,
        "client": ApolloClient {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "clearStoreCallbacks": Array [],
          "clientAwareness": Object {},
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": ApolloLink {
            "request": [Function],
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "cacheKeyRoot": CacheKeyNode {
                  "children": null,
                  "key": null,
                },
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "ssrMode": false,
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "cacheKeyRoot": CacheKeyNode {
                  "children": null,
                  "key": null,
                },
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "typeDefs": undefined,
          "version": "2.5.0",
          "watchQuery": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
